### ISSUE TITLE
[Design] Create a Contract

### ISSUE THREAD

**Issue by ninamak** [HHS/OPRE-OPS#208]
_This is a draft.  To be validated with PO.  My current thinking is that we should support an earlier stage of contract creation than is provided for here, rather than postponing that for later, so that we don't optimize designs for a workflow that we'll be discarding._

### User Story
As a Budget Officer, I need to enter contracts that have been awarded so that OPRE staff know what money has been obligated and when they can start invoicing against the contract.

### Background
- A contract goes through many stages (see [Lifecycle of a Contract](https://app.mural.co/t/gsa6/m/gsa6/1626978306340/7b440640a5a2244ef06ac28f34f0347df76d2e67?sender=ninamak9326)).
  - OPRE documents the contracts they're planning to execute
  - OPRE documents which specific acquisition actions they're planning to take that year (all or part of a planned contract)
  - OPRE submits an initial requisition to start the acquisition/procurement process
  - OPRE awards a contract
  - OPRE submits a final requisition to commit and obligate the money
  - Money is obligated
  - Money is certified
- It is only after the contract is awarded and money is obligated and certified that OPRE can officially start spending money (invoicing against the contract).
- This is the stage at which we want users to be able to enter contract info into OPS (future work will address earlier stages).

### Who needs to be able to create contracts?
- Budget Officers
  - Future stories could enable CORs to be able to enter contract info and submit to Budget Officers for approval

### When do they need to be able to create contracts?
- After the contract is awarded and money is obligated and certified
  - Future stories could support entering contract info at earlier stages in the lifecycle

### What data do they need to create about the contract?
- Contract info
  - Contract Name
  - Contract Number
  - Contract Type (single-select: defined list)
  - COR
  - Project Lead
  - Project Team
  - POP Start Date
  - POP End Date
  - Status
  - **Note:** Need to include some sort of amount(s) here, which may end up being calculated fields based off of budget lines.

- Procurement info
  - Proc Shop Name (single-select)
  - Proc Shop Contract Specialist (short text)
  - Proc Shop Fee (calculated field, currency)
  - Task Order Number 
  - PO Number. 
  - Acquisition Vehicle (**Note:** This is called Acquisition Type elsewhere)
  - Product Service Code (**Question:** What is this??)
  - **Note:** Several of these fields are currently listed under Contract Info in MAPS, but is this a more appropriate home?

- Vendor info
  - Vendor Name (type-ahead, single-select, with option to create a new vendor)
 
- Budget Lines
  - User should be able to add one or more Budget Lines to a contract
    - FY
    - Name/Description and/or some way to associate Budget Line to CLIN
    - CAN
    - Amount
    - Status
    - Funds Expire On
    - **Questions:**
      - Line Description seems to repeat CLIN #/CLIN Name, so what purpose does it serve?
      - What is Invoice Line # for?
      - When do you care about Mod # and Date Obligated and POP Starts/Ends?
      - How helpful is Time to Complete?
      - What is a logical order to enter/display these fields?

### What associated info do they need to create?
- Proc Shop info
  - Proc Shop Name (short text)
  - Proc Shop Fee Type (single-select: Flat Fee or Percent Fee)
  - Proc Shop Fee Rate (percent with 2 decimals; automatically disable or pre-populate with 0% if Flat Fee)

- Vendor info
  - Vendor Name
  - Vendor DUNS
  - Vendor Address
  - Vendor Contact
  - Vendor Contact Phone
  - Vendor Contact Email
  - Vendor Head of Contracts
  - Vendor Head of Contracts Phone
  - Vendor Head of Contracts Email

- Staff info
  - First Name
  - Last Name
  - Title
  - Federal employee?  (or select whether Federal employee or Contractor)
  - COR Certification (one or more fields to track what COR certifications this staff member has)
  - Phone
  - Email

### What info has historically been collected and we need to determine if it is no longer needed or will be captured in a different place?
- New Project (Contract) Template
  - Why a new contract
  - If a re-compete, who is incumbent
  - Acquisition/Contract Type
    - **Question:** What is the difference between Acquisition/Contract Type and Contract Type?
  - Acquisition strategy
  - Competition type
  - State where bulk of work will be done
  - Contract description
  - NAICS code
  - Program support code
  - COR/POC title, phone, email
    - **Question:** Is it more important to see contact info for the COR or the Project Lead or both?
  - Budget Lines: Month & Year (not just FY)
  - Sub-total contract amount current FY
  - Total contract amount

**Note:** Will also need to reference existing data tables to see if any other data needs to be captured.

**Note:** Will also need to reference any other MAPS pages that show contract info (e.g., grids or reports).
